frontend/src/components/Seleccion.js

```diff
@@ -0,0 +1,34 @@
+import React from "react";
+
+class Seleccion extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { gastos: [], value: " " };
+  }
+
+  render() {
+    return (
+      <form>
+        <select
+          value={this.state.value}
+          onChange={this.props.listadoDeGastoPorMes}
```

Comment:
De alguna manera le tenés que pasar esta prop al componente. Lo más fácil es que este componente esté dentro de `Gasto` y de ahí le pasás la función. Además, deberías llamarla con un parámetro (el mes seleccionado), quedaría así:
```suggestion
          onChange={() => this.props.listadoDeGastoPorMes(this.state.value)}
```